pkg/cloudprovider/providers/azure/azure_test.go

```diff
@@ -359,29 +626,244 @@ func TestReconcileSecurityWithSourceRanges(t *testing.T) {
+	pip, err = az.reconcilePublicIP(testClusterName, &svcUpdated, true /* wantLb*/)
+	if err != nil {
+		t.Errorf("Unexpected error: %q", err)
+	}
+	validatePublicIP(t, pip, &svcUpdated, true)
+
+	// Update to internal service again
+	pip, err = az.reconcilePublicIP(testClusterName, &svc, true /* wantLb*/)
+	if err != nil {
+		t.Errorf("Unexpected error: %q", err)
+	}
+	validatePublicIP(t, pip, &svc, true)
+}
+
+func getTestCloud() (az *Cloud) {
+	az = &Cloud{
 		Config: Config{
-			TenantID:          "tenant",
-			SubscriptionID:    "subscription",
-			ResourceGroup:     "rg",
-			Location:          "westus",
-			VnetName:          "vnet",
-			SubnetName:        "subnet",
-			SecurityGroupName: "nsg",
-			RouteTableName:    "rt",
+			TenantID:                     "tenant",
+			SubscriptionID:               "subscription",
+			ResourceGroup:                "rg",
+			VnetResourceGroup:            "rg",
+			Location:                     "westus",
+			VnetName:                     "vnet",
+			SubnetName:                   "subnet",
+			SecurityGroupName:            "nsg",
+			RouteTableName:               "rt",
+			PrimaryAvailabilitySetName:   "asName",
+			MaximumLoadBalancerRuleCount: 250,
 		},
 	}
+	az.operationPollRateLimiter = flowcontrol.NewTokenBucketRateLimiter(100, 100)
+	az.LoadBalancerClient = NewFakeAzureLBClient()
+	az.PublicIPAddressesClient = NewFakeAzurePIPClient(az.Config.SubscriptionID)
+	az.SubnetsClient = NewFakeAzureSubnetsClient()
+	az.SecurityGroupsClient = NewFakeAzureNSGClient()
+	az.VirtualMachinesClient = NewFakeVirtualMachinesClient()
+	az.InterfacesClient = NewFakeInterfacesClient()
+
+	return az
+}
+
+const networkInterfacesIDTemplate = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/networkInterfaces/%s"
+const primaryIPConfigIDTemplate = "%s/ipConfigurations/ipconfig"
+
+// returns the full identifier of a publicIPAddress.
+func getNetworkInterfacesID(subscriptionID string, resourceGroupName, nicName string) string {
+	return fmt.Sprintf(
+		networkInterfacesIDTemplate,
+		subscriptionID,
+		resourceGroupName,
+		nicName)
+}
+
+// returns the full identifier of a private ipconfig of the nic
+func getPrimaryIPConfigID(nicID string) string {
+	return fmt.Sprintf(
+		primaryIPConfigIDTemplate,
+		nicID)
+}
+
+const TestResourceNameFormat = "%s-%d"
+const TestVMResourceBaseName = "vm"
+const TestASResourceBaseName = "as"
+
+func getTestResourceName(resourceBaseName string, index int) string {
+	return fmt.Sprintf(TestResourceNameFormat, resourceBaseName, index)
+}
+
+func getVMName(vmIndex int) string {
+	return getTestResourceName(TestVMResourceBaseName, vmIndex)
+}
+
+func getASName(az *Cloud, vmIndex int, numAS int) string {
+	asIndex := vmIndex % numAS
+	if asIndex == 0 {
+		return az.Config.PrimaryAvailabilitySetName
+	}
+
+	return getTestResourceName(TestASResourceBaseName, asIndex)
+}
+
+func getNICName(vmIndex int) string {
+	// test supporting on 1 nic per vm
+	return getVMName(vmIndex)
```

Comment:
this is confusing - it appears to return the VM name - is it that in the fake situation the NICs have the same names as the VMs that contain them?  If so maybe comment that more clearly

Reply:
added comment